packages/cli/mock/src/runMockServer.ts

```diff
@@ -37,33 +36,95 @@ export class MockServer {
         this.app.use(express.json({ limit: "50mb", strict: false }));
         this.port = port;
 
+        // Map of endpoint path to map of HTTP method to endpoints that fall into both
+        const endpointGroups: Map<string, Map<string, HttpEndpoint[]>> = new Map();
+
         for (const service of Object.values(ir.services)) {
             for (const endpoint of service.endpoints) {
                 const endpointPath = getFullPathForEndpoint(endpoint);
                 context.logger.debug(`Registering ${endpoint.method} ${endpointPath} ...`);
-                switch (endpoint.method) {
+
+                const existingGroup = endpointGroups.get(endpointPath) ?? new Map();
+                const existingEndpoints = existingGroup.get(endpoint.method) ?? [];
+
+                endpointGroups.set(endpointPath, existingGroup.set(endpoint.method, [...existingEndpoints, endpoint]));
+            }
+        }
+
+        const listGroups = Array.from(endpointGroups);
+        const sortedEndpoints = listGroups.sort(this.sortExpressPaths);
+
+        for (const [endpointPath, methodToEndpoints] of sortedEndpoints) {
+            for (const [method, endpoints] of methodToEndpoints) {
+                switch (method) {
                     case "GET":
-                        this.app.get(endpointPath, getRequestHandler(endpoint));
+                        this.app.get(endpointPath, getRequestHandler(endpoints));
                         break;
                     case "POST":
-                        this.app.post(endpointPath, getRequestHandler(endpoint));
+                        this.app.post(endpointPath, getRequestHandler(endpoints));
                         break;
                     case "PUT":
-                        this.app.put(endpointPath, getRequestHandler(endpoint));
+                        this.app.put(endpointPath, getRequestHandler(endpoints));
                         break;
                     case "PATCH":
-                        this.app.patch(endpointPath, getRequestHandler(endpoint));
+                        this.app.patch(endpointPath, getRequestHandler(endpoints));
                         break;
                     case "DELETE":
-                        this.app.delete(endpointPath, getRequestHandler(endpoint));
+                        this.app.delete(endpointPath, getRequestHandler(endpoints));
                         break;
-                    default:
-                        assertNever(endpoint.method);
                 }
             }
         }
     }
 
+    // Sort the paths with a priority for literal path parts over parameters
+    // e.g. /movie/:movieId should come after /movie/123, such that express
+    // can match the exact endpoint before the variable one (/movie/:movieId)
+    sortExpressPaths(
```

Comment:
nit: Should we mark this `private`?
```suggestion
    private sortExpressPaths(
```